test/rewards/AutoCompoundingRewards.ts

```diff
@@ -778,8 +860,45 @@ describe('AutoCompoundingRewards', () => {
                         it('should distribute all tokens', async () => {
                             await autoCompoundingRewards.setTime(programEndTime[distributionType]);
                             for (let i = 0; i < Math.ceil(setups.length / AUTO_PROCESS_REWARDS_COUNT); i++) {
+                                const tokenAmountToDistributeArray: BigNumber[] = new Array<BigNumber>(
+                                    AUTO_PROCESS_REWARDS_COUNT
+                                );
+                                const poolTokenAmountToBurnArray: BigNumber[] = new Array<BigNumber>(
+                                    AUTO_PROCESS_REWARDS_COUNT
+                                );
+                                const remainingRewards: any[] = new Array<BigNumber>(AUTO_PROCESS_REWARDS_COUNT);
+                                for (let j = 0; j < AUTO_PROCESS_REWARDS_COUNT; j++) {
+                                    const index = AUTO_PROCESS_REWARDS_COUNT * i + j;
+                                    if (index == tokens.length) {
+                                        break;
+                                    }
+                                    const prgramData = await autoCompoundingRewards.program(tokens[index].address);
+                                    const { tokenAmountToDistribute, poolTokenAmountToBurn } = await getRewards(
+                                        prgramData,
+                                        tokens[index],
+                                        rewardsMath,
+                                        new TokenData(setups[index].tokenSymbol),
+                                        setups[index].tokenSymbol === TokenSymbol.BNT ? bntPool : externalRewardsVault
+                                    );
+                                    tokenAmountToDistributeArray[j] = tokenAmountToDistribute;
+                                    poolTokenAmountToBurnArray[j] = poolTokenAmountToBurn;
+                                    remainingRewards[j] = prgramData.remainingRewards;
+                                }
                                 const res = await autoCompoundingRewards.autoProcessRewards();
-                                await expect(res).to.emit(autoCompoundingRewards, 'RewardsDistributed');
+                                for (let j = 0; j < AUTO_PROCESS_REWARDS_COUNT; j++) {
+                                    const index = AUTO_PROCESS_REWARDS_COUNT * i + j;
+                                    if (index == tokens.length) {
+                                        break;
+                                    }
+                                    await expect(res)
+                                        .to.emit(autoCompoundingRewards, 'RewardsDistributed')
+                                        .withArgs(
```

Comment:
I know that you did it before my comment, but FYI - you really don't need the exact rewards data to check that a program for TKN1 was processed, while for TKN2 it was not. You only need to check the first parameter.

Reply:
Fro. my experimenting, once you add `withArgs`, it expects to find **all** the parameters, though perhaps I did something wrong while experimenting. In any case, it now verifies all of the event parameters.

Reply:
This is why I keep emphasizing again and again that you should test for this argument **manually** (i.e., without using `to.emit`, but checking `{ events }` directly)

Reply:
Yeah, verifying the rewards data (when it's actually unneeded) makes the tests much more complicated and harder to follow.